wandb/data_types.py

```diff
@@ -1051,8 +1101,41 @@ def bind_to_run(self, *args, **kwargs):
                 kwargs["id_"] = "{}{}".format(id_, i) if id_ is not None else None
                 self._masks[k].bind_to_run(*args, **kwargs)
 
-    def to_json(self, run):
-        json_dict = super(Image, self).to_json(run)
+    def to_json(self, run_or_artifact):
+        from wandb.sdk import wandb_run
+        from wandb.sdk import wandb_artifacts
+
+        if isinstance(run_or_artifact, wandb_run.Run):
+            run = run_or_artifact
+            json_dict = super(Image, self).to_json(run)
+        elif isinstance(run_or_artifact, wandb_artifacts.Artifact):
+            artifact = run_or_artifact
+            if self._classes is None:
+                raise ValueError(
```

Comment:
Why do we force Image objects in artifacts to have class labels? My understanding of artifacts is that it is a general "versioned folder". This constraint might not align with unsupervised ML cases such as a "baby face filter" (see social media). In such a case, classes/labels might not make sense.